modules/services/window-managers/awesome.nix

```diff
@@ -6,7 +6,7 @@ let
 
   cfg = config.xsession.windowManager.awesome;
   awesome = cfg.package;
-  getLuaPath = lib: dir: "${lib}/${dir}/lua/${pkgs.luaPackages.lua.luaversion}";
+  getLuaPath = lib: dir: "${lib}/${dir}/lua/${awesome.lua.luaversion}";
```

Comment:
I would like to avoid out-of-tree reimplementation of lua utilities such as getLuaPath here. I would like to use lua flat installs in the future but that requires packages not to replicate their own lua wrapping code https://github.com/NixOS/nixpkgs/issues/56398#issuecomment-912096841 . 

I have some ongoing work to improve the lua ecosystem https://github.com/NixOS/nixpkgs/pull/177556/files#diff-558d30a16316ebf76080fb16d8b82a341313e48617e224e2acf30b5f64952990 but until then could you use https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/lua-modules/lib.nix#L39 instead ?

Reply:
Maybe we can merge this as-is for now and do the improvement in a future PR?

Reply:
Yeah, while I appreciate the effort of reducing duplicate code/boilerplate when dealing with Lua that you seem to be working on, this PR is a oneline fix and just mirrors the same fix I made in nixpkgs a little while ago.

I am totally out of context for the Lua ecosystem improvements you're making and a jump from 'oneline fix' to 'use the entirely new better lua system' was a bit too much for me so I forgot about this PR :sweat_smile: